spec/lib/dynamic_validations_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# rubocop:disable Lint/ConstantDefinitionInBlock
+RSpec.describe 'DynamicValidations' do
+    context 'when added, includes the dynamic validations' do
```

Comment:
I would move this whole block to the Batch model spec as it is essentially testing batch behaviour.

Reply:
Moving this again causes an intermittent failure. However, I think this test will become redundant when [Y24-040](https://github.com/sanger/sequencescape/issues/4068) is done because it will have the necessary classes and would not require us to stub any class. So I think we can remove this test after the next story is done.